aten/src/ATen/core/function_schema.h

```diff
@@ -402,6 +409,12 @@ struct TORCH_API FunctionSchema {
   // FunctionSchema::may_contain_alias will include that functionality.
   bool may_alias(const SchemaArgument& lhs, const SchemaArgument& rhs) const;
 
+  // Returns whether lhs and rhs may alias directly or whether lhs/rhs are a container
+  // that may contain elements that alias the other argument.
+  // bidirectional = false only returns whether lhs may contain an alias of rhs
+  // while bidirectional = true returns both directions.
+  bool may_contain_alias(const SchemaArgument& lhs, const SchemaArgument& rhs, bool bidirectional = true) const;
+
```

Comment:
What is the use case for `bidirectional = false` ?

Reply:
It's there if you want more specific information on aliasing. For instance if you have a Tensor[] and a Tensor then may_contain_alias(Tensor[], Tensor, false) might return true, while may_contain_alias(Tensor, Tensor[], false) will never return true.